week-4/E-arrays-of-objects/exercise-2.js

```diff
@@ -7,50 +7,85 @@ Each destination has a name, a distance from Glasgow, and a list of transportati
 3) Print in the console all the destination names more than 300 kms far away and reachable by train.
 */
 
-
 var destination1 = {
-    destinationName: "Edinburgh",
-    distanceKms: 80,
-    transportations: ["car", "bus", "train"]
+  destinationName: "Edinburgh",
+  distanceKms: 80,
+  transportations: ["car", "bus", "train"]
 };
 
 var destination2 = {
-    destinationName: "London",
-    distanceKms: 650,
-    transportations: ["car", "bus", "train"]
+  destinationName: "London",
+  distanceKms: 650,
+  transportations: ["car", "bus", "train"]
 };
 
 var destination3 = {
-    destinationName: "Paris",
-    distanceKms: 900,
-    transportations: ["train", "plane"]
+  destinationName: "Paris",
+  distanceKms: 900,
+  transportations: ["train", "plane"]
 };
 
 var destination4 = {
-    destinationName: "Dublin",
-    distanceKms: 350,
-    transportations: ["plane", "ferry"]
+  destinationName: "Dublin",
+  distanceKms: 350,
+  transportations: ["plane", "ferry"]
 };
 
-var travelDestinations = [destination1, destination2, destination3, destination4];
+var travelDestinations = [
+  destination1,
+  destination2,
+  destination3,
+  destination4
+];
 
 /*
 DO NOT EDIT ANYTHING ABOVE THIS LINE
 WRITE YOUR CODE BELOW
 */
+var destinationNamesWithin500Kms = travelDestinations
+  .filter(function(element) {
+    return element.distanceKms <= 500 ? true : false;
+  })
+  .map(function(element) {
+    return element.destinationName;
+  });
+// Complete here
+var destinationNameReachableByFerry = travelDestinations
+  .filter(reachableByFerry =>
+    reachableByFerry.transportations.includes("ferry")
+  )
+  .map(reachableByFerry => reachableByFerry.destinationName);
+//var destinationNameReachableByFerry = travelDestinations
```

Comment:
Best not to leave commented out code around, but the code that isn't commented out is all good.

Reply:
ok thanks i will take care of that.